EventListener/ControllerListener.php

```diff
@@ -50,13 +50,18 @@ public function __construct(Reader $reader)
      */
     public function onKernelController(FilterControllerEvent $event)
     {
-        if (!is_array($controller = $event->getController())) {
+        if (!is_array($controller = $event->getController()) && !is_object($controller)) {
             return;
         }
 
-        $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
-        $object = new \ReflectionClass($className);
-        $method = $object->getMethod($controller[1]);
+        if (is_array($controller)) {
+            $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
+            $object = new \ReflectionClass($className);
+            $method = $object->getMethod($controller[1]);
+        } else {
+            $object = new \ReflectionClass(get_class($controller));
```

Comment:
getting the class name should use the ClassUtils too, to account for proxy classes (JMSDiExtraBundle generates such proxies for instance).

@Taluu no, this would get the wrong method for proxy objects (the current code is also broken)